### PR TITLE
Feat/user can see all events created by himself

### PR DESCRIPTION
**Pull request by Denisa1128**

### CODE REVIEW

File: src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java [hunk 1]
```diff
@@ -3,22 +3,39 @@
...
-    public List<EventDTO> getUsersEvents() {
-        return null;
+    public List<EventDTO> getUserEvents(Long id) {
+        Optional<User> userWithEvent = userRepository.findById(id);
+        if(!userWithEvent.isPresent()) {
+            throw new RuntimeException("User with ID " + id + " not found");
+        }
+
+        User user = userWithEvent.get();
+        List<Event> events = eventRepository.findByUserId(user.getId());
```
Comment (alinaitschool) [inline src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java]: You can replace eventRepository.findByUserId(user.getId()) with user.getManagedEvent

File: src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java [hunk 2]
```diff
@@ -3,22 +3,39 @@
...
+
+        return events.stream()
```
Comment (alinaitschool) [inline src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java]: I like the fact you use a functional implementation for this

File: src/main/java/com/itschool/eventmanagment/controllers/UserController.java [hunk 3]
```diff
@@ -18,7 +18,7 @@ public UserController(UserService userService) {
     }
 
     @GetMapping("api/users/{id}/events")
-    public ResponseEntity<List<EventDTO>> getUserEvents(@PathVariable String id) {
-        return ResponseEntity.ok(userService.getUsersEvents());
+    public ResponseEntity<List<EventDTO>> getUserEvents(@PathVariable Long id) {
```
Comment (alinaitschool) [inline src/main/java/com/itschool/eventmanagment/controllers/UserController.java:21]: Thank you take into consideration my suggestion regarding naming

File: src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java (hunk 1 above)
Comment (Denisa1128) [reply src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java]: Done

File: src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java (hunk 2 above)
Comment (Denisa1128) [reply src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java]: Thanks

File: src/main/java/com/itschool/eventmanagment/controllers/UserController.java (hunk 3 above)
Comment (Denisa1128) [reply src/main/java/com/itschool/eventmanagment/controllers/UserController.java:21]: Thanks

**Comment by alinaitschool** [comment]
@Denisa1128 I approve you PR and you can merge it. Good job!

File: src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java (hunk 1 above)
Comment (alinaitschool) [reply src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java]: @Denisa1128 maybe you forgot to push the changes. I can't find it

File: src/main/java/com/itschool/eventmanagment/repositories/EventRepository.java [hunk 4]
```diff
@@ -4,6 +4,9 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface EventRepository extends JpaRepository<Event, Long > {
+    List<Event> findByUserId(Long id);
```
Comment (alinaitschool) [inline src/main/java/com/itschool/eventmanagment/repositories/EventRepository.java]: @Denisa1128 please take a close look here. As a hint you can check it and compare with EventRepository from dev branch.

**Review by alinaitschool — APPROVED** [review]

**Comment by alinaitschool** [comment]
@Denisa1128 also tested and works as required. Big like for your implementation

**Pull request merged**